web3/_utils/async_transactions.py

```diff
@@ -52,7 +52,10 @@ async def _estimate_gas(w3: "Web3", tx: TxParams) -> Awaitable[int]:
 
 
 async def _gas_price(w3: "Web3", tx: TxParams) -> Awaitable[Optional[Wei]]:
-    return await w3.eth.generate_gas_price(tx) or w3.eth.gas_price  # type: ignore
+    gas_price = await w3.eth.generate_gas_price(tx)  # type: ignore
+    if gas_price is None:
```

Comment:
See my comment below, I think this can probably be back to a one-liner:

```py
return await w3.eth.generate_gas_price(tx)
```

Looking a bit further I think we really only support `gas_price_strategy` as static values for Async so I also think that we can remove the need to `await` here. I did some quick testing and that does seem to be the case locally but we can create a separate issue to track that since that deviates from this PR. I would be content with just the suggested change from above here for now.

Reply:
Not sure about removing `async`, since we are passing tx object inside which might get used in some async calls. So I'm not sure whether it worth removing this convenience, given how simple it is.

P.S.: These are just my thoughts, I have not experienced async part of web3 yet